tock/utilization/unit.py

```diff
@@ -10,51 +10,77 @@
 # Building context for display of business unit level utilization data
 def unit_billing_context(unit):
     """Build context dict for utilization by unit"""
-    last_week = _get_unit_billing_data(unit)
-    last_month = _get_unit_billing_data(unit, recent_periods=settings.RECENT_TIMECARDS_FOR_BILLABILITY)
-    this_fy = _get_unit_billing_data(unit, fiscal_year=True)
+
+    # all users that have ever created a timecard for this unit
+    users = User.objects.filter(timecards__unit=unit).distinct()
+
+    last_week = _get_unit_billing_data(users, unit)
+    last_month = _get_unit_billing_data(users, unit, recent_periods=settings.RECENT_TIMECARDS_FOR_BILLABILITY)
+    this_fy = _get_unit_billing_data(users, unit, fiscal_year=True)
 
     context = _build_utilization_context(last_week, last_month, this_fy)
 
+    usernames_without_hours = [e.username for e in _employees_without_hours(users, last_week['data'] + last_month['data'] + this_fy['data'])]
```

Comment:
Could this be simplified if we assume that `this_fy` is the boundary of the reporting for `unit_billing_context`, then doing that filtering in the `User` Queryset above?

Reply:
I think I see what you mean - `last_week` and `last_month` timecards are subsets of `this_fy` timecards for a given user, so if we additionally filter the QS on line 15 to timecards dated within this FY, that would give us the set of `employees_with_hours`

I think... 🤔

Reply:
Yes - this works, and is simpler 👍 

I have to update some of the tests around this to make them a bit more date-resilient (they were setup with hardcoded RP dates, but the FY is determined by looking at the system clock). Should be ready for review by tomorrow AM.